packages/wallet/src/services/HandlesTracker.ts

```diff
@@ -30,7 +44,38 @@ const handleInfoEquals = (a: HandleInfo, b: HandleInfo) =>
   deepEquals(a.tokenMetadata, b.tokenMetadata) &&
   deepEquals(a.nftMetadata, b.nftMetadata);
 
-export const createHandlesTracker = ({ tip$, assetInfo$, handlePolicyIds, logger, utxo$ }: HandlesTrackerProps) =>
+const getHandleFromAssetId = (assetId: Cardano.AssetId) => {
+  const assetName = Cardano.AssetId.getAssetName(assetId);
+  const assetNameContent = Asset.AssetNameLabel.decode(assetName)?.content;
+  return Buffer.from(assetNameContent || assetName, 'hex').toString('utf8');
+};
+
+const hydrateHandleAsync =
```

Comment:
It would be great to test this separately since the existing tests are stubbing it

Reply:
Fixed here: https://github.com/input-output-hk/cardano-js-sdk/pull/786/commits/ebd4cbb66d8b5992a06fae1bc8d0221567abf06a